File: problem20.cpp
#include <stdio.h>
#include "BigInteger.h"
#include <string>
#include <sstream>

/*
Find the sum of the digits in the 
number 100!
*/
using namespace std;

string digitSum(string number) {
	if (!BigInteger::isStrNum(number))
		return 0;
	string sum = "0";
	for (int i = 0; i < number.length(); i++) 
		sum = BigInteger::addStr(sum, number.substr(i,1));
	return BigInteger::trimZeroes(sum);
}
 
int main(int argc, char** argv) {
	string huge = BigInteger::factorialStr("100");
	printf("100!: %s\n", huge.c_str());
	printf("num digits: %s\n", digitSum(huge).c_str());
	return 0;
}

File: problem20.py
#!/usr/bin/env python

'''
find the sum of the digits 
in the number 100!
'''

def fac(n):
	if n == 0 or n == 1:
		return 1;
	return n * fac(n-1)

def sum_dig(n):
	n_str = str(n)
	rolling_sum = 0
	for i in n_str:
		rolling_sum += int(i)
	return rolling_sum

if __name__ == "__main__":
	print "10: " + str(sum_dig(fac(10)))
	print "100: " + str(sum_dig(fac(100)))	
	
